### PR TITLE
Task vv/132 delete podcast functionality

### PR DESCRIPTION
**Pull request by vitalik-prog**

### CODE REVIEW

File: backend/src/services/episode/episode.service.ts [hunk 1]
```diff
@@ -245,6 +245,13 @@ class Episode {
 
     return episode;
   }
+
+  public async deleteAllByPodcastId(id: number): Promise<void> {
+    const episodes = await this.#episodeRepository.getAllByPodcastId(id);
+    episodes.forEach((episode) => {
+      this.delete(episode.id);
+    });
```
Comment (artem-totality) [inline backend/src/services/episode/episode.service.ts:253]:
```suggestion
    episodes.forEach(async (episode) => {
      await this.delete(episode.id);
    });
```
We should await before all episodes will be deleted and only after move on

File: backend/src/services/podcast/podcast.service.ts [hunk 2]
```diff
@@ -186,6 +194,36 @@ class Podcast {
...
+    const podcast = await this.#podcastRepository.getById(id);
+
+    if (!podcast) {
+      throw new HttpError({
+        status: HttpCode.NOT_FOUND,
+        message: ErrorMessage.PODCAST_NOT_FOUND,
+      });
+    }
+
+    await this.#podcastRepository.delete(id);
```
Comment (artem-totality) [inline backend/src/services/podcast/podcast.service.ts]: We should delete podcast after all episodes wil be deleted. But before the images deleting.

File: backend/src/services/podcast/podcast.service.ts [hunk 3]
```diff
@@ -186,6 +194,36 @@ class Podcast {
...
+
+    const episodes = await this.#episodeService.getAllByPodcastId(id);
+    const isEpisodesExist = Boolean(episodes);
+
+    if (isEpisodesExist) {
+      await this.#episodeService.deleteAllByPodcastId(id);
+    }
```
Comment (artem-totality) [inline backend/src/services/podcast/podcast.service.ts:213]:
```suggestion
      await this.#episodeService.deleteAllByPodcastId(id);
```
We don't need this checks, if there are no episodes episodeService.deleteAllByPodcastId(id) will do nothing.

File: backend/src/services/episode/episode.service.ts (hunk 1 above)
Comment (artem-totality) [inline backend/src/services/episode/episode.service.ts]:
```suggestion
    for (let i = 0; i < episodes.length; i++) {
      await this.delete(episodes[i].id);
    }
```
We should await before all episodes will be deleted, and only then move on

Comment (what1s1ove) [reply backend/src/services/episode/episode.service.ts]: Good catch, but we can do it better
```tsx
await Promise.all(episodes.map((it) => this.delete(it.id)))
```

File: backend/src/services/podcast/podcast.service.ts (hunk 2 above)
Comment (what1s1ove) [reply backend/src/services/podcast/podcast.service.ts]: Good catch 👍

File: frontend/src/common/types/podcast/delete-action-podcast-payload.type.ts [hunk 4]
```diff
@@ -0,0 +1,6 @@
+type DeleteActionPodcastPayload = {
```
Comment (what1s1ove) [inline frontend/src/common/types/podcast/delete-action-podcast-payload.type.ts:1]: not sure that we need it
we can take id of episode from params 🤔

**Review by what1s1ove — CHANGES_REQUESTED** [review]

File: backend/src/services/podcast/podcast.service.ts (hunk 3 above)
Comment (vitalik-prog) [reply backend/src/services/podcast/podcast.service.ts:213]: I think this condition will not allow additional access to the database

File: frontend/src/common/types/podcast/delete-action-podcast-payload.type.ts (hunk 4 above)
Comment (vitalik-prog) [reply frontend/src/common/types/podcast/delete-action-podcast-payload.type.ts:1]: I also use it for identifying userId for custom redirect to user page.

Comment (what1s1ove) [reply frontend/src/common/types/podcast/delete-action-podcast-payload.type.ts:1]: Good!

**Review by what1s1ove — APPROVED** [review]

**Pull request merged**